player-with-speedbar/player.js

```diff
@@ -0,0 +1,80 @@
+const player = document.querySelector(".player");
+const video = player.querySelector(".viewer");
+const progress = player.querySelector(".progress");
+const progressBar = player.querySelector(".progress__filled");
+const toggleButton = player.querySelector(".toggle");
+const skipButtons = player.querySelectorAll("[data-skip]");
+const volumeRange = player.querySelector(".player__slider");
+const speed = document.querySelector('.speed');
+const speedBar = document.querySelector('.speed__bar');
+
+progressBar.style.flexBasis = 0;
+
+// Stop-play video & Update Icon
+function checkPlaying() {
+  if (!video.paused) {
+    video.pause();
+    toggleButton.textContent = "►";
+  }
+  else {
+    video.play();
+    toggleButton.textContent = "❚ ❚";
+  }
+}
+
+function rebutVideo() {
+  video.currentTime = 0;
+  video.pause();
+  toggleButton.textContent = "►";
+}
+
+// Skip 5 secs
+function skip() {
+  video.currentTime += parseInt(this.dataset.skip);
+}
+
+// Set Volume
+function handleRangeUpdate() {
+  video[this.name] = this.value;
```

Comment:
i guess this is from the example. but please be careful ith this in such functions